test/test_torch.py

```diff
@@ -8334,6 +8334,100 @@ def test_contiguous(self, device):
         x.set_(x.storage(), 0, x.size(), stride)
         self.assertTrue(x.is_contiguous())
 
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float, torch.double, torch.half, torch.int, torch.long, torch.cfloat, torch.cdouble, torch.bool, torch.uint8)
+    def test_tensor_split_sections(self, device, dtype):
+        input_sizes = [
+            (0,),
+            (10,),
+            (10, 0),
+            (0, 10),
+            (4, 10),
+            (12, 3),
+        ]
+        for input_size in input_sizes:
+            if dtype.is_floating_point or dtype.is_complex:
+                a = torch.randn(*input_size, dtype=dtype, device=device)
+            else:
+                if dtype == torch.bool:
+                    a = torch.randint(2, input_size, dtype=dtype, device=device)
+                else:
+                    a = torch.randint(0, 127, input_size, dtype=dtype, device=device)
+            a_n = a.cpu().numpy()
+            for dim in range(a.dim()):
```

Comment:
This test looks pretty good. Would you try it with the tensor you construct and a transposed version of it, too, to test the operation on discontinguous tensors?

Reply:
Also let dim=-1